h5py/h5t.pyx

```diff
@@ -949,24 +954,18 @@ cdef class TypeFloatID(TypeAtomicID):
         e_bias = self.get_ebias()
 
         # Handle non-standard exponent and mantissa sizes.
-        if m_size > 112 or (2**e_size - 2 - e_bias) > 16383 or (1 - e_bias) < -16382:
-            raise ValueError('Invalid exponent or mantissa size in ' + str(self))
-        elif m_size > 52 or (2**e_size - 2 - e_bias) > 1023 or (1 - e_bias) < -1022:
-            size = 16
-        elif m_size > 23 or (2**e_size - 2 - e_bias) > 127 or (1 - e_bias) < -126:
-            size = 8
-        elif m_size > 10 or (2**e_size - 2 - e_bias) > 15 or (1 - e_bias) < -14:
-            size = 4
-        elif not hasattr(np, 'float16'):
-            # This build doesn't have float16; promote to float32
-            size = 4
+        for size, finfo in sorted(available_ftypes.items()):
+            nmant = finfo.nmant
+            if nmant == 63 and finfo.nexp == 15:
+                # This is an 80-bit float, correct mantissa size
+                nmant += 1
```

Comment:
Could be because of the integer bit in https://en.wikipedia.org/wiki/Extended_precision#x86_extended_precision_format, @mraspaud can you confirm this?